src/src/Helpers/TarStream.cpp

```diff
@@ -0,0 +1,603 @@
+#include "../Helpers/TarStream.h"
+
+#if FEATURE_TARSTREAM_SUPPORT
+# include "../Globals/ESPEasy_time.h"
+# include "../Helpers/ESPEasy_Storage.h"
+# include "../Helpers/StringConverter.h"
+
+/**
+ * TarStream : Create/receive a .tar file while streaming via http webserver
+ * Copyright (c) 2023.. Ton Huisman for ESPEasy
+ *
+ * Changelog: See TarStream.h
+ */
+
+/**
+ * TarFileInfo_struct implementation
+ */
+TarFileInfo_struct::TarFileInfo_struct(const String _fileName,
+                                       size_t       _fileSize) :
```

Comment:
I am more used to have `_foo` (underscore prefixed) as class member names.
You're using this reversed, but also a bit inconsistent.
This can be a bit tricky if the compiler decides to use the move operator for member initialization in the `foo(_foo)` construction phase.
Not sure if C++ will ever do this without having the variable declared using `&&` as move reference.

Reply:
Didn't want to add the overhead of getters/setters for such a small struct, and dislike having the members prefixed with underscores. I'll rename the constructor arguments to avoid the underscores.